src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -140,12 +142,108 @@ public ResultWrapper<Address> eth_coinbase()
             return ResultWrapper<UInt256?>.Success(0);
         }
 
-        [Todo("Gas pricer to be implemented")]
-        public ResultWrapper<UInt256?> eth_gasPrice()
+        private bool add_transactions_from_block_to_set(Block block, ref SortedSet<UInt256> sortedSet, UInt256 finalPrice, UInt256? ignoreUnder = null, long? maxCount = null)
         {
-            return ResultWrapper<UInt256?>.Success(20.GWei());
+            ignoreUnder ??= UInt256.Zero;
+            int length = block.Transactions.Length;
+            int added = 0;
+            if (length > 0)
+            {
+                for (int index = 0; index < length && (maxCount == null || sortedSet.Count < maxCount); index++)
+                {
+                    Transaction transaction = block.Transactions[index];
+                    if (transaction.GasPrice >= ignoreUnder)
+                    {
+                        sortedSet.Add(transaction.GasPrice);
+                        added++;
+                    }
+                }
+
+                switch (added)
+                {
+                    case 0:
+                        sortedSet.Add(finalPrice);
+                        return (false);
+                    case 1:
+                        return (false);
+                    default:
+                        return true;
+                }
+            }
+            else
+            {
+                sortedSet.Add(finalPrice);
+                return false;
+            }
+        }
+
+        private void latest_gas_price(long headBlockNumber, long genesisBlockNumber, ref UInt256? latestGasPrice)
+        {
+            while (headBlockNumber >= genesisBlockNumber) //should this be latestBlock or headBlock?
+            {
+                Transaction[] transactions = _blockFinder.FindBlock(headBlockNumber)!.Transactions;
```

Comment:
My understanding is that we should filter out EIP1559 transactions because their gas price field would be invalid. Please add a test case for that scenario.

Reply:
Please check Peter comment: https://github.com/ethereum/pm/issues/328#issuecomment-853234014